### PR TITLE
OptimizeInstructions: Eq64 of 0 => Eqz64

### LINKED ISSUE #2417: OptimizeInstructions and i64
It appears that the `OptimizeInstructions` pass does not yet hand-optimize i64 instructions as much as it optimizes i32 instructions. This recently [surfaced](https://github.com/AssemblyScript/assemblyscript/pull/933#discussion_r341818922) when we changed compilation of `!someI64` to emit `eqz(isTrueish)` instead of using a dedicated `isFalseish` helper, leading to missed optimizations like

```wat
i64.const 0
i64.ne
i32.eqz
```

becoming

```wat
i64.const 0
i64.eq
```

but not being optimized further to

```wat
i64.eqz
```

initially leading to the assumption that this is caused by the expression collapsing very late, thus requiring `--converge`, which is not the case at it seems. Pinning it here so we don't forget :)

### PR DESCRIPTION
**Pull request by kripken**
Fixes #2417 

cc @dcodeIO

### CODE REVIEW

File: src/passes/OptimizeInstructions.cpp [hunk 1]
```diff
@@ -1312,6 +1312,8 @@ struct OptimizeInstructions
                    !EffectAnalyzer(getPassOptions(), binary->left)
                       .hasSideEffects()) {
           return binary->right;
+        } else if (binary->op == EqInt64) {
+          return Builder(*getModule()).makeUnary(EqZInt64, binary->left);
```
Comment (tlively) [inline src/passes/OptimizeInstructions.cpp]: Where is this optimization done for `i32.eqz`? I would expect this optimization to be organized to be symmetrically located with the `i32` version.

Comment (dcodeIO) [reply src/passes/OptimizeInstructions.cpp]: The other one appears to be here: https://github.com/WebAssembly/binaryen/blob/7fa8c78dc02501ba4cd6088b3664a1eb8b794cda/src/passes/OptimizeInstructions.cpp#L373 also handling various sign extension cases, which made me shy away from a quick fix as it seems there's more to it.

Comment (kripken) [reply src/passes/OptimizeInstructions.cpp]: Yeah, this is not structured in a very symmetrical way atm. It grew organically + some thoughts about checking common patterns early. So i32.eqz is much more common than i64.eqz, and hence they are differently located. It's very possible that's not worth it, and this should be refactored.

Comment (tlively) [reply src/passes/OptimizeInstructions.cpp]: OK, LGTM for now.

File: test/passes/O.bin.txt [hunk 2]
```diff
@@ -51,9 +49,10 @@
   )
   (loop $label$3
    (if
-    (i64.ne
-     (local.get $0)
-     (i64.const 0)
+    (i32.eqz
+     (i64.eqz
```
Comment (MaxGraey) [inline test/passes/O.bin.txt]: It seems `i32.eqz i64.eqz` could be eliminate at all for CFG operands

Comment (kripken) [reply test/passes/O.bin.txt]: What do you mean by "CFG" here?

Comment (MaxGraey) [reply test/passes/O.bin.txt]: I meant arguments for control operations like `if (i64)` but I forgot that `if` should be accept only `i32` input argument, so series of `eqz` may be better than `i64.ne` + `i64.const 0`

**Pull request merged**